### PR TITLE
Feature/meshstructured

### PR DESCRIPTION
**Pull request by vincentchabannes**
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/feelpp/feelpp/pulls) for the same change?
- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your changes?
- [x] Have you successfully run the Feel++ testsuite with your changes locally?
- [ ] Have you written Doxygen comments in your contribution ?

-----

MeshStructured features :
- 1d/2d/3d with hypercube element (TODO : Simplex)
- geometry currently available: `hyperrectangle`. The setup of the geometry is  given by a json file, see  examples in test_meshstructured. (TODO : other shapes as for example an arbitrary Quadrangle(2d) or QuadrilaterallyFacedHexahedron(3d) by using geo transformation of hypercube).
- partitioning currently available: `striped`  one direction (TODO : by block)

@prudhomm It will break some code that uses MeshStructured : 
- template parameters of class MeshStructured
- constructor args

### CODE REVIEW

File: testsuite/feeldiscr/test_meshstructured.cpp [hunk 1]
```diff
@@ -25,279 +25,247 @@
...
+
+FEELPP_ENVIRONMENT_NO_OPTIONS
 
-namespace Feel
-{
 
-inline
-AboutData
-makeAbout()
+#if 0
```
Comment (prudhomm) [inline testsuite/feeldiscr/test_meshstructured.cpp:56]: the test does not work with coordinates ?

**Review by prudhomm — APPROVED** [review]
Is there a test for the case with coordinates ? I found one but it is commented out

**Comment by prudhomm** [comment]
nice job for refactoring this

File: testsuite/feeldiscr/test_meshstructured.cpp (hunk 1 above)
Comment (vincentchabannes) [reply testsuite/feeldiscr/test_meshstructured.cpp:56]: the coordinate test is [here](https://github.com/feelpp/feelpp/blob/feature/meshstructured/testsuite/feeldiscr/test_meshstructured.cpp#L213-L269). This comment is a hbf test but it was always commented before my change.

**Pull request merged**